packages/utils/safe-promise.js

```diff
@@ -1,10 +1,10 @@
 'use strict';
 
-const withResult = (p) =>
+const safePromise = (p) =>
     new Promise((resolve) => {
         p.then((result) => resolve([null, result])).catch((err) =>
-            resolve([err, null])
+            resolve([err])
```

Comment:
Because you removed the `null`, should we also remove the `null` from L5 to make it consistent? At the moment, err is either present or `null` and the result is either present or `undefined`. I find this a little confusing.

Keeping in mind nullish coalescing and [logical assignment operators](https://medium.com/better-programming/the-top-3-new-javascript-es-2021-es-12-features-im-excited-about-a3ac129efbb2#19d1) I think it's a much stronger API to be able to interrogate both indexes of the tuple with the same logic (i.e. `null` or `undefined`, but not both).

Reply:
@allanchau, we don't need to worry about this. Apparently `null` and `undefined` are both _nullish_ values.